doc/sphinx/src/models.rst

```diff
@@ -69,41 +168,101 @@ in the :doc:`EOS API <using-eos>`.
+|                          |                      | Internal Energy          |
 +--------------------------+----------------------+--------------------------+
 | :math:`B_S(\rho, T)`     | Isentropic Bulk      | Density, Temperature     |
 +--------------------------+ Modulus              +--------------------------+
-| :math:`B_S(\rho, e)`     |                      | Density, Internal Energy |
+| :math:`B_S(\rho, e)`     |                      | Density, Specific        |
+|                          |                      | Internal Energy          |
 +--------------------------+----------------------+--------------------------+
 | :math:`\Gamma(\rho, T)`  | Gruneisen Parameter  | Density, Temperature     |
 +--------------------------+                      +--------------------------+
-| :math:`\Gamma(\rho, e)`  |                      | Density, Internal Energy |
+| :math:`\Gamma(\rho, e)`  |                      | Density, Specific        |
+|                          |                      | Internal Energy          |
 +--------------------------+----------------------+--------------------------+
 
 A point of note is that "specific" implies that the quantity is intensive on a
-per unit mass basis. It should be assumed that the internal energy is *always*
-specific since we are working in terms of density (the inverse of specific
-volume).
+per unit mass basis. It should be assumed that the internal energy and entopry
+are *always* specific since we are working in terms of density (the inverse of
+specific volume).
 
-Disambiguation
-````````````````
+Entropy availability
+````````````````````
+For an arbitrary equation of state, a change in entropy in terms of temperature
+and volume is given by
 
-Gruneisen Parameter
-'''''''''''''''''''
+.. math::
+
+    \Delta S = \int\limits_{S_0}^{S} \mathrm{d}S =
+        \int\limits_{T_0}^{T} \left(\frac{\partial S}{\partial T}\right)_V
+        \mathrm{d}T
+        + \int\limits_{V_0}^{V} \left(\frac{\partial S}{\partial V}\right)_T
+        \mathrm{d}V,
+
+which can be simplified using a definition of the heat capacity and a Maxwell
+relation to become
+
+.. math::
+
+    \Delta S =
+        \int\limits_{T_0}^{T} \frac{C_V}{T} \mathrm{d}T
+        + \int\limits_{V_0}^{V} \left(\frac{\partial P}{\partial T}\right)_V
+        \mathrm{d}V.
+
+Similarly, expressing the entropy in terms of *energy* and volume yields
+
+.. math::
+
+    \Delta S =
+        \int\limits_{e_0}^{e} \frac{1}{T(\rho, e)} \mathrm{d}e
+        + \int\limits_{V_0}^{V} \frac{P(\rho, e)}{T(\rho, e)}
+        \mathrm{d}V
+
+after substituting for the appropriate derivatives using the first law of
+thermodynamics.
+
+Importantly for the analytic EOS, these integrals will tend to diverge as the
+temperature and volume approach zero if the heat capacity does not also
+approach zero. This necessitates appropriate choices for the reference
+states :math:`T_0` and :math:`V_0`.
+
+.. note::
```

Comment:
Maybe make this a warning instead of a note?

Reply:
I had been reserving the warnings for EOS that just plain don't provide entropy, which is why I chose notes here. My thinking was that for EOS where entropy isn't enabled, your code will crash if you try to access it, so that's more serious.

Reply:
Sure that's fair.